app/models/katello/upstream_consumer.rb

```diff
@@ -0,0 +1,61 @@
+require 'katello/resources/candlepin'
+
+module Katello
+  class UpstreamConsumer < OpenStruct
+    CP_CONSUMER = Resources::Candlepin::UpstreamConsumer
+
+    class << self
+      def bind_entitlements(organization, params)
+        CP_CONSUMER.organization = organization
+        params.fetch(:pools).map do |pool|
+          JSON.parse(CP_CONSUMER['entitlements'].post(nil, params: map_pool_attributes(pool)))
+        end
+      ensure
+        CP_CONSUMER.organization = nil
+      end
+
+      def get_entitlements(organization, params)
+        CP_CONSUMER.organization = organization
+        JSON.parse(CP_CONSUMER['entitlements'].get(params: cp_request_params(params)))
+      ensure
+        CP_CONSUMER.organization = nil
+      end
+
+      def map_pool_attributes(pool)
+        if pool
```

Comment:
style comment: I think it's more Ruby to use `return unless pool` as the first line of this method instead of surrounding the method body with an `if` - just my opinion and I'm not necessarily asking that you change it.